### DISCUSSION TITLE
2022-07-06 Meeting Agenda, Recording & Notes

### DISCUSSION THREAD

**Discussion by andy-blum** [ddev/ddev#3800] [DDEV Advisory Group]
Previous meeting: #3651 
Next Meeting: #3963 

## **Agenda**

- Turn on recording, ask for a minutes-taker
- New test runners deployed
- Upstream donations status
- Financial status
- "Bus Factor" status and path forward
- Questions:
  - More nominations for upstream projects to donate to with 10% of GitHub Sponsors
- Anything on anybody's mind!

## Minutes

- Recording started
- New test runners deployed
  - This is thanks to community support!
  - Retired old laptops that were being used for tests. Replaced by mini PCs and 4 mac minis, 3 amd64 and 1 arm64. Mac Stadium also donated a mac mini in their colo.
  - Still three laptops running WSL2.
  - Been thinking about ways to make all the Windows machines do both WSL2 or traditional Windows.
- On Upstream Donations:
  - List is in comments below
  - We're close to the 10% level
  - Been thinking about how far Composer has come compared to Composer 1. Seemed like a good upstream contribution for us.
  - Mkdocs Material is being used for the docs site.
  - Andrew Berry: How do we prevent circular chains of donations, where no one wins except platform vendors? Should we publicize our contributions more prominently?
    - Randy: Yes, but not sure it's not that big a threat right now.
    - GitHub isn't even taking credit card fees right now, so worst case it wouldn't lead to lost funds. Not seeing any evidence of surcharge. That may happen if an organization sponsors a project.
- Financial Status:
  - See the income statement and balance sheet in comments below
  - $3200 in GitHub sponsor income
  - Mac Stadium sorted out sponsoring the test runner. They had sponsored the wrong thing, and they refunded the money we were charged. Beyond generous!
  - Equipment purchases seem to be done for now.
  - About $5K in the bank
  - Paypal was complicated with so many different transactions.
- "Bus Factor" and path forward
  - Randy is a critical contributor to the project, much is dependent on him, and there's more work than Randy can do alone. Lots of scope expansion!
  - Would like to diversify the ownership of some of the scope.
  - Thoughts and nominations?
  - Could we have a campaign to come up with more money? But, the real issue is finding people.
  - Had a call with a contributor, but that person didn't have spare time. Employer might though? But, unavailable in the end.
  - How do we find people with talent and interest?
    - For example: maintaining test runners, improving tests. Our tests are very intense.
    - Or new features, like supporting GitHub code spaces.
    - Andrew Belcher:
      - 1: Figuring out who would pick up development and support if Randy is not contributing anymore.
      - 2: Keys and access to the repository, organization, etc.
        - Could we have some people to be trusted who can have keys in case of disaster?
        - Randy: Fabian Potencier has full access to drud in GitHub.
        - However, tokens for things like pushing to AUR, API keys, etc. Those are in GitHub as secrets. Perhaps get out via a GitHub action.
        - Doesn't make sense to share LastPass shares with a random person?
          - Then again… we could give access to two others? Andrew and Ofer who already use LastPass?
          - **Action Item:** Randy will work on this.
    - Truls: Make a private team on GitHub? Then we can put in links with keys and so on.
      - Randy will make sure to document.
      - Once we add members to the org, the costs of running the org go up.
      - Ofer Shaal: We can use Bitwarden as an OSS password manager too.
      - Randy: Would rather not change from LastPass.
      - But on GitHub access: Having a team of 3-4 raises the cost on the org. I've thought about it.
      - Isn't it free for open source? Depends on the plan, and so it would actually be free for right now. Worried about the upsell. The cost jumps when the team grows beyond a certain point. Want to avoid having to remove memberships just to keep costs contained.
      - With Fabien being an owner, it's not likely a big issue. Important part is knowing Fabien has this access and how we'd contact him if needed.
    - Graham: Can you share LastPass without costs? Randy says so.
  - Mike Barkas: Is there a list of major components or things others could start taking over?
    - Testing is the biggest area. Maintaining the test runners, and maintaining the tests themselves.
    - We run two tests on every platform and architecture. Perhaps we could run some tests on Linux only? We need to know that the basics work on all the other architectures. But, worried these may not be interesting to the average contributor.
    - Would love thoughts on how we could scope out things, like find a Windows expert? Simon Gilbert from the Typo 3 community is a real Windows user and contributor, but his availability is limited. Having platform-specific experts would be a real win.
    - A big initiative: improving the install and UX for WSL2. WSL2 is really great, but it's a long list of steps to install it. People stumble setting up, but love it when they get it running. 13 steps is too many.
    - Andrew Belcher: Enjoys test automation, but not a ton of time.
    - Nico: Also interested, but may not be available until the Winter.
      - Randy: Will give Andrew an overview.
    - Nico: Plan to automate ddev + phpstorm with WSL2, would like to be included in this topic. Need to simplify for other IDE users too.
    - Randy spends a good amount of time reviewing and restarting tests on Windows with Docker Desktop. They have fails on the average PR. Regular developers shouldn't need this expertise. This situation can be improved.
    - Could give access to someone to the Windows and macOS test runners.
    - What happens is issues like Windows docker desktop not coming up, such as docker daemon (service?) on Windows crashing. Need to restart host or Docker.
    - Someone maintaining test runners will need to pay attention to the test failures.
    - Truls: Do we have test runner monitoring?
      - Randy: Just the tests themselves.
      - We could have an agent to report to a monitoring panel.
      - Truls: We could also monitor how the runner is working, performance. For example, DataDog to get information on crashes, report logs, etc.
- More nominations for upstream projects to donate to with 10% of GitHub Sponsors
- Anything else?
  - Truls: Possible to post on the blog asking if someone wants to see how ddev works, give a guided tour of ddev development?
    - Randy: Let's do this!
  - Nico: Can ddev be easier to use for larger companies? Enterprise certificates, msi's to distribute. It would be of interest to my company.
    - Need to write some hooks to trust our enterprise TLS certificate. Perhaps a way to use mkcert to use an enterprise certificate.
    - We need to trust the cert inside the ddev container, and every host OS has a different way of managing certificates.
    - Not a clear concept, so there's no PR yet.
    - Randy: Should be able to use custom certs too, see the docs!
  - Ofer Shaal:
    - Microsoft is pushing devcontainers. Gitpod should support it soon. Can we make ddev work with them?
    - Randy: See [https://plugins.jetbrains.com/plugin/18813-ddev-integration](https://plugins.jetbrains.com/plugin/18813-ddev-integration) ! Thank you Nico for the work.

## Action items (20220708 edit by rfay)
* Test infrastructure
  - [ ] Add monitoring for docker status, etc.
  - [ ] Consider nightly reboot
* Spreading disaster recovery capability
  - [ ] Disaster plan
  - [ ] Shared critical keys with trusted community members (with docs that go with them) (Andrew Belcher and Ofer Shaal volunteered)
  - [ ] Plan with @fabpot for sharing drud repo admin access if the need arises
* Explore areas where ownership could be diversified
  - [ ] Test infrastucture and its maintenance (monitoring?), especially mac and Windows (nico might be interested in winter)
  - [ ] Tests themselves, making them less wasteful but still effective
* Publicize and promote a guided tour of DDEV development, testing, maintenance, support (blog + tweets + action)

## Recording

Recording: https://www.dropbox.com/s/encka7cc6mxp81u/20220706%20DDEV%20Advisory%20Group%20Meeting.mp4?dl=0

**Comment by rfay** [comment]
Financial Statements:

[Localdev Foundation Balance Sheet 2022-07-05-19_21.pdf](https://github.com/drud/ddev/files/9048950/Localdev.Foundation.Balance.Sheet.2022-07-05-19_21.pdf)
[Localdev Foundation Profit and Loss 2022-07-05-19_20.pdf](https://github.com/drud/ddev/files/9048951/Localdev.Foundation.Profit.and.Loss.2022-07-05-19_20.pdf)

**Comment by rfay** [comment]
Upstream contributions:

- PHP Foundation - Open Collective - $30/month
- Patreon - Colima - $30/month
- shaal/Drupalpod - $5/month
- oerdnj/deb.sury.org - $10/month
- derickr/xdebug - $10/month
- homebrew - $10/month
- composer - $25/month (via drud)
- squidfunk/markdown-material $10/month (via rfay, less complicated)

Total $130/month

**Comment by rfay** [comment]
Issue for an emergency plan and shared knowledge of secrets:
* https://github.com/drud/ddev/issues/3962
